ArduPlane/mode_LoiterAltQLand.cpp

```diff
@@ -10,17 +10,12 @@ bool ModeLoiterAltQLand::_enter()
         return true;
     }
 
+    // If we were already in a loiter then use that waypoint. Else, use the current point
+    const Location loiter_wp = plane.nav_controller->reached_loiter_target() ? plane.next_WP_loc : plane.current_loc;
```

Comment:
```suggestion
    const Location loiter_wp = (plane.nav_controller->reached_loiter_target() && !plane.nav_controller->data_is_stale()) ? plane.next_WP_loc : plane.current_loc;
```
Doing the stale check here makes it much to prove there is no unintended change for other modes.

Reply:
applied